src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -291,5 +291,26 @@ public override Uri GetUri(string name)
             }
             return !(await source.ExistsAsync());
         }
+
+        public ICloudBlockBlob GetCloudBlockBlobReference(string name)
+        {
+            Uri uri = ResolveUri(name);
+            string blobName = GetName(uri);
+            CloudBlockBlob blob = _directory.GetBlockBlobReference(blobName);
+
+            if (blob.Exists())
```

Comment:
>Exists [](start = 21, length = 6)

There's an async version of `Exists`, can you use that?

Reply:
Good catch.